mephisto/abstractions/providers/mturk/mturk_utils.py

```diff
@@ -61,21 +61,40 @@ def check_aws_credentials(profile_name: str) -> bool:
 def setup_aws_credentials(
     profile_name: str, register_args: Optional[DictConfig] = None
 ) -> bool:
+    if not os.path.exists(os.path.expanduser("~/.aws/")):
+        os.makedirs(os.path.expanduser("~/.aws/"))
+    aws_credentials_file_path = "~/.aws/credentials"
+    expanded_aws_file_path = os.path.expanduser(aws_credentials_file_path)
     try:
         # Check existing credentials
         boto3.Session(profile_name=profile_name)
         if register_args is not None:
             # Eventually we could manually re-parse the file and see
             # if the credentials line up or not, then fix ourselves
-            print(
-                f"WARNING credentials provided, but there's already a "
-                f"profile for {profile_name}. If these don't line up, you'll "
-                f"need to manually navigate to your ~/.aws/credentials file "
-                f"and remove the entry for this profile name, then run again.\n"
-                f"As this profile is currently loading, we consider it "
-                f"successfully registered anyways."
-            )
+            aws_credentials_file_string = ""
+            with open(expanded_aws_file_path, "r") as aws_credentials_file:
+                aws_credentials_file_string = aws_credentials_file.read()
+            # accessing the aws_credentials_file
+            aws_credentials = aws_credentials_file_string.split("\n")
+            # iterating to get the profile
+            profileFound = False
+            for credentialIndex in range(0, len(aws_credentials)):
+                if profileFound:
+                    aws_credentials[credentialIndex] = "aws_access_key_id={}\n".format(
+                        register_args.access_key_id)
+                    aws_credentials[credentialIndex+1] = "aws_access_key_id={}\n".format(
+                        register_args.access_key_id)
+                    break
+                if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
+                    profileFound = True
+
+                with open(expanded_aws_file_path, "w") as aws_credentials_file:
+                    # overWrite login details
+                    aws_credentials_file.write("\n".join(aws_credentials))
+                logger.warning(f"The provided keys does not match for profile {profile_name}"
+                               f"As a result of which we're updating the credentials, overwriting ones that already existed for the profile ")
```

Comment:
Note, we now override always, and aren't checking to see if we've actually made any changes. A valid message would then be:
```
logger.warning(
    f"We found an existing entry for {profile_name}. As new credentials have been provided, "
    f"we're updating the credentials, overwriting ones that already existed for the profile "
)
```